src/quick-lint-js/parse.h

```diff
@@ -1441,12 +1441,41 @@ class parser {
     QLJS_WARNING_IGNORE_GCC("-Wshadow-local")
 
     std::optional<identifier> last_ident;
+    const char8 *async_token_begin = nullptr;
     function_attributes method_attributes = function_attributes::normal;
+    bool async_static = false;
 
     auto parse_and_visit_field_or_method_impl =
-        [this, &v](std::optional<identifier> property_name,
-                   source_code_span property_name_span,
-                   function_attributes method_attributes) -> void {
+        [this, &v, &async_static, &async_token_begin](
+            std::optional<identifier> property_name,
+            source_code_span property_name_span,
+            function_attributes method_attributes) -> void {
+      if (async_static && !this->peek().has_leading_newline) {
+        switch (this->peek().type) {
+        case token_type::star:
+          method_attributes = function_attributes::generator;
+          this->skip();
+          [[fallthrough]];
```

Comment:
Blocking: I think fallthrough after skipping is dangerous. What if `*` is followed by `+`, for example? I think that'd cause a crash in `identifier_name()` on line 1470.